src/core/a-entity.js

```diff
@@ -211,15 +212,26 @@ var proto = Object.create(ANode.prototype, {
 
   load: {
     value: function () {
+      var self = this;
+
       if (this.hasLoaded) { return; }
+
       // Attach to parent object3D.
       this.addToParent();
+
+      // Scene load.
+      function sceneLoadCallback () { self.updateComponents(); }
       if (this.isScene) {
-        ANode.prototype.load.call(this, this.updateComponents.bind(this));
-      } else {
-        ANode.prototype.load.call(this, this.updateComponents.bind(this),
-                                  function (el) { return el.isEntity; });
+        ANode.prototype.load.call(this, sceneLoadCallback);
+        return;
+      }
+
+      // Entity load.
+      function entityLoadCallback () {
```

Comment:
The function gets in the way of understanding the main body of the function. I want to see what the function does furst and maybe later want to see what happens asynchronously when the entity loads.